### PR TITLE
Adding code to prevent user from deleting recurring event and refactoring code for cached calendar service parameters

### PR DESCRIPTION
**Pull request by mkieres**

### CODE REVIEW

File: device_calendar/lib/src/common/constants.dart [hunk 1]
```diff
@@ -2,5 +2,5 @@ class Constants {
   static const String fromJsonMapIsNull = 'The json object is null';
 
   static const String invalidArgument = 'Provided arguments are invalid';
-  static const String createEventArgumentReuirements = 'To create an event you must provide calendar with an id, event with title, start date and end date';
+  static const String createOrUpdateEventArgumentReuirements = 'To create or update an event you must provide calendar with an id, event with title, start date and end date (where start date must be before end date)';
```
Comment (nickrandolph) [inline device_calendar/lib/src/common/constants.dart]: createOrUpdateEventArgumentRe**q**uirements <- missing letter

File: device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarService.kt [hunk 2]
```diff
@@ -34,35 +34,30 @@ import android.provider.CalendarContract.Events
...
 import com.builttoroam.devicecalendar.common.ErrorCodes.Companion.EVENTS_RETRIEVAL_FAILURE
 import com.builttoroam.devicecalendar.common.ErrorCodes.Companion.EVENT_CREATION_FAILURE
 import com.builttoroam.devicecalendar.common.ErrorMessages.Companion.CREATE_EVENT_ARGUMENTS_NOT_VALID_MESSAGE
+import com.builttoroam.devicecalendar.common.ErrorMessages.Companion.DELETING_RECURRING_EVENT_NOT_SUPPORTED_MESSAGE
 import com.builttoroam.devicecalendar.common.ErrorMessages.Companion.EVENTS_START_DATE_LARGER_THAN_END_DATE_MESSAGE
+import com.builttoroam.devicecalendar.models.CalendarMethodsParametersCacheModel
 import java.util.*
 
 
 public class CalendarService : PluginRegistry.RequestPermissionsResultListener {
```
Comment (MaikuB) [inline device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarService.kt]: I think it's worth renaming this class to be CalendarDelegate similar to the image picker plugin as a service has a different meaning in Android https://developer.android.com/guide/components/services

File: device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/common/ErrorCodes.kt [hunk 3]
```diff
@@ -8,5 +8,6 @@ class ErrorCodes {
         const val EVENTS_RETRIEVAL_FAILURE: String = "events_retrieval_failure";
         const val EVENT_CREATION_FAILURE: String = "event_creation_failure";
         const val CALENDAR_IS_READ_ONLY: String = "calendar_is_read_only";
+        const val DELETING_RECURRING_EVENT_NOT_SUPPORTED: String = "deteling_recurring_event_not_supported";
```
Comment (MaikuB) [inline device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/common/ErrorCodes.kt]: typo in your string value

File: device_calendar/lib/src/common/constants.dart [hunk 4]
```diff
@@ -2,5 +2,5 @@ class Constants {
   static const String fromJsonMapIsNull = 'The json object is null';
 
   static const String invalidArgument = 'Provided arguments are invalid';
-  static const String createEventArgumentReuirements = 'To create an event you must provide calendar with an id, event with title, start date and end date';
+  static const String createOrUpdateEventArgumentRequirements = 'To create or update an event you must provide calendar with an id, event with title, start date and end date (where start date must be before end date)';
```
Comment (MaikuB) [inline device_calendar/lib/src/common/constants.dart]: do they actually pass a calendar object with it's id as well or they pass a a calendar id? if it's the latter then the text should change to more accurately reflect that

File: device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarService.kt (hunk 2 above)
Comment (mkieres) [reply device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarService.kt]: Yeah, good point - don't like how it works in Android though

File: device_calendar/lib/src/common/constants.dart (hunk 4 above)
Comment (mkieres) [reply device_calendar/lib/src/common/constants.dart]: Just an id, will change the text

File: device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/common/ErrorCodes.kt (hunk 3 above)
Comment (mkieres) [reply device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/common/ErrorCodes.kt]: ty

File: device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarService.kt (hunk 2 above)
Comment (MaikuB) [reply device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarService.kt]: i know you don't :)

File: device_calendar/lib/src/common/constants.dart (hunk 4 above)
Comment (mkieres) [reply device_calendar/lib/src/common/constants.dart]: Actually, we were passing the entire Calendar object. I changed that to accept only an Id

**Pull request merged**